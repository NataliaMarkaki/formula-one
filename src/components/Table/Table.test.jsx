import * as React from "react";
import { act } from "react-dom/test-utils";
import { render, unmountComponentAtNode } from "react-dom";

describe("Table", () => {
  let Table;
  let container = null;
  const mockDrivers = [
    {
      givenName: "Alexander",
      familyName: "Albon",
      permanentNumber: "23",
      nationality: "Thai",
      dateOfBirth: "1996-03-23"
    }
  ];
  jest.mock("./TableRow", () => props => <div>{JSON.stringify(props)}</div>);

  beforeAll(() => {
    Table = require(".").default;
  });

  beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
  });

  afterEach(() => {
    jest.clearAllMocks();
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });

  it("should render drivers table", () => {
    act(() => {
      render(<Table drivers={mockDrivers} />, container);
    });
    expect(container).toMatchSnapshot();
  });
});
