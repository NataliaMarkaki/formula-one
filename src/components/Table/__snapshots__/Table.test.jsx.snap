// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table should render drivers table 1`] = `
.c0 {
  margin: 20px;
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: transparent;
  color: #323232;
  box-shadow: 0 0 2px #c9d3d7;
}

<div>
  <div
    class="c0"
  >
    <div>
      {"name":"Alexander Albon","permanentNumber":"23","nationality":"Thai","DOB":"1996-03-23","isOdd":true}
    </div>
  </div>
</div>
`;
